src/functions/indGraphSparse.jl

```diff
@@ -0,0 +1,56 @@
+struct IndGraphSparse{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  F::Base.SparseArrays.CHOLMOD.Factor{T} #LDL factorization
+end
+
+function IndGraphSparse(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  K = [speye(n) A'; A -speye(m)]
+
+  F = LinAlg.ldltfact(K)
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  return IndGraphSparse(m, n, A, F)
+end
+
+# is_convex(f::IndGraph) = true
+# is_set(f::IndGraph) = true
+# is_cone(f::IndGraph) = false
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphSparse,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+  res = [c + f.A' * d; zeros(f.m)]
+  xy = f.F \ res
+  x[:] = xy[1:f.n]
+  y[:] = xy[end - f.m + 1:end]
```

Comment:
I did some tests on this line with n=500.
The `copy!` function only does a simple check for the dimensions and then calls an efficient C function that loops over the elements and adds them, allocating almost nothing. `@benchmark` from BenchmarkTools estimates :
79.601 ns and 0 bytes allocated. Memory scales as O(1)

The code  `x .= xy[1:n]`  or `x[:] = y[1:n]` allocates a temporary vector for the result of `xy[1:n]` and then copies  that to `x`, while doing bounds checks in both stages.
Takes 527.463 ns, allocates 4.09 KiB. Memory scales as O(n)

 `x[:] .= xy[1:n]` is similar but also creates a view of x first, seemingly slowing it down considerably.
Takes 2.791 μs and 4.47 KiB. Memory scales as O(n)

`@views  x .= xy[1:n]` or equivalent `x .= view(y,1:n)` creates a view of 1:n in y and then does efficient copy.
Takes 1.916 μs and 592 bytes. Memory scales as O(1)

`view(x,1:n) .= view(y,1:n)` and `view(x,:) .= view(y,1:n)`
Takes roughly 3.5 μs with 1 KiB memory, but scales as O(1) in memory.